src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -3902,6 +3902,68 @@ void verifyParsedAsRecord()
             }
         }
 
+        [Fact, CompilerTrait(CompilerFeature.RecordStructs)]
+        public void RecordParsing_Ref()
+        {
+            var text = "ref record R;";
+            UsingTree(text, options: TestOptions.RegularPreview);
```

Comment:
consider laying these out in sequential order, i.e. 8, 9, then preview, or possibly putting them in their own test methods.